### PR TITLE
Implementation of Protocol ABC and his derivate. 

### PR DESCRIPTION
**Pull request by alberto-cardini**

### CODE REVIEW

**Comment by leonardo-guglielmi** [comment]
Evaluate whether to use DMA or interrupts in the future for non-blocking communications (depending on the volume of data to be written on the memories)

**Review by leonardo-guglielmi — APPROVED** [review]

**Comment by danimorga00** [comment]
In the SPI protocol, do we want to leave the responsibility of raising and lowering the chip selects outside the class? So the caller has to do it?

**Comment by danimorga00** [comment]
Doing so doesn't make sense: we would like to standardize the interfaces but then in practice the caller must know what communication the sensor has to raise and lower the CS. The best thing would be to distinguish between a physical communication and an e2e, the physical one keeps track of the physical channel and calls the HAL transmit, the e2e one instead has a physical communication and adds the CS in the SPI case and maybe the address in the i2c case.

File: .github/workflows/cmake-multi-platform.yml
Comment (niccocco) [inline .github/workflows/cmake-multi-platform.yml:1]: Why are we using matrix compile to test on different OSes?
Is this just a template or is there a reasoning behind?

Comment (alberto-cardini) [reply .github/workflows/cmake-multi-platform.yml:1]: Is an Action provided by github to test the building suit on different platforms. Usually is added to the project to ensure that the committed files build properly on any OS.

**Pull request closed without merging**